tools/wt_verify.py

```diff
@@ -38,24 +38,118 @@
+separator = "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=\n"
+
+def output_pretty(output):
     """
-    Visualizing data
+    Print output for parsing script
     """
-    pass
+    str = "{"
+    for i, checkpoint in enumerate(output):
+        if i:
+            str += ","
+        str += "\n\t" + checkpoint + ": {"
+        for j, key in enumerate(output[checkpoint]):
+            value = output[checkpoint][key]
+            if j:
+                str += ","
+            str += "\n\t\t" + key + ": {\n\t\t\t" + json.dumps(value)
+            if "data" in value:
+                str += "\n\t\t\t\"data\": " + json.dumps(value["data"]) 
+            if "hs_modify" in value:
+                str += "\n\t\t\t\"hs_modify\": " + json.dumps(value["hs_modify"]) 
+            if "hs_update" in value:
+                str += "\n\t\t\t\"hs_update\": " + json.dumps(value["hs_update"]) 
+            str += "\n\t\t}"
+        str += "\n\t}"
+    str += "\n}"
+    return str
+
+
+def is_int(string):
+    """
+    Check if string can be converted to an integer
+    """
+    if isinstance(string, str):
+        try :
+            return int(string)
+        except ValueError:
+            return string
 
 
-def parse_output(output):
+def string_to_iterable(line):
+    """
+    Helper that converts string to dictionaries and / or lists
     """
-    Parse the output file
+    dict = {}
+    for x in line.split(" | "):
+        kv_pair = x.split(": ", 1)
+        if kv_pair[1][0] == "[" and kv_pair[1][-1] == "]":
+            kv_pair[1] = kv_pair[1][1:-1].split(", ")
+            kv_pair[1] = list(map(lambda n: is_int(n), kv_pair[1]))
+        if kv_pair[0] == "addr": 
+            temp = kv_pair[1][0].split(": ")
+            dict.update({"object_id": is_int(temp[0]), "offset_range": temp[1], "size": is_int(kv_pair[1][1]), 
+                        "checksum": is_int(kv_pair[1][2])})
+        else:
+            dict[kv_pair[0]] = is_int(kv_pair[1])
+    return dict
+
+
+def parse_node(f, line, output, checkpoint_name, cur_node_id):
+    cur_node = {}
+    line = line[len("- "):-1]
+    page_type = ([line.split(": ")[0]] + line.split(": ")[1].split())[-1]
+    if page_type == "root":
+        cur_node_id = is_int(line.split(": ")[0])
+        cur_node["page_type"] = page_type
+    elif page_type == "internal" or page_type == "leaf":
+        cur_node_id = is_int(line.split(": ")[0])
+    else:
+        pass
+    line = f.readline()
+    while line and line != separator and not line.startswith("- "):
+        if line.startswith("\t> "): # metadata for new node
+            cur_node |= string_to_iterable(line[len("\t> "):-1])  
+        line = f.readline()
+    output[checkpoint_name][cur_node_id] = cur_node 
+    return [cur_node_id, line]
+
+
+def parse_output(file_path):
     """
+    Parse the output file of dump_pages
+    """
+    with open(file_path, "r") as f:
+        output = {}
+        line = f.readline() # separator
```

Comment:
I do not think the comment is necessary here.

Reply:
fixed https://github.com/wiredtiger/wiredtiger/pull/10164/commits/5ddd16a669636a7946308b778e0e35d50d90966c